src/App.tsx

```diff
@@ -1,26 +1,179 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  createTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { FilterType } from './types/FilterType';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { ErrorType } from './types/ErrorType';
+import { TodoList } from './components/TodoList';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorType>(
+    ErrorType.Default,
+  );
+  const [filter, setFilter] = useState<FilterType>(FilterType.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+
+  const filteredTodos = todos.filter(todo => {
+    switch (filter) {
+      case FilterType.All:
+        return true;
+      case FilterType.Completed:
+        return todo.completed;
+      case FilterType.Active:
+        return !todo.completed;
+      default:
+        return false;
+    }
+  });
+
+  const notCompletedTodosCount = todos.filter(todo => !todo.completed).length;
+  const allTodosAreCompleted = useMemo(
+    () => todos.every(todo => todo.completed),
+    [todos],
+  );
+
+  const inputNameRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorType.LoadTodo);
+      });
+  }, []);
```

Comment:
useEffect should be at the bottom, before rendering.
Thus - all useEffects will be in one place and you can easily manage them